<launch>
    <!-- if rosbag_play == true -->
    <arg name="topicname_lidar"     default="/debug/pc_raw"/> <!-- "/velodyne_points/lidar0"/> -->
    <arg name="topicname_pose"      default="/ct_icp/pose/odom"/> 
    <!--ALOAM: /aft_mapped_to_init-->
    <!--CT-ICP /ct_icp/pose/odom-->

    <arg name="dataset_name"        default="CARLA"/>   <!-- KITTI // CARLA -->
    <arg name="rosbag_play"         default="true"/>    <!-- true: rosbag // false: pcd data -->
    <arg name="T01_slam"            default="true"/>   <!-- true: T01 from slam // false: T01 from GT -->
    <arg name="data_number"         default="03"/>      <!-- KITTI data: 00 01 02 05 07 // CARLA data: 01 01_001 01_002 03 03_001 03_002--> 

    <arg name="dataset_kitti_dir"         default="/mnt/e/KITTI_odometry/"/>
    <arg name="dataset_carla_dir"         default="/mnt/h/mapless_dataset/CARLA/"/>

    <node pkg="mapless_moving" name="mapless_moving_node" type="mapless_moving_node" output="screen">
        <param name="topicname_lidar" type="string" value="$(arg topicname_lidar)"/>
        <param name="topicname_pose" type="string" value="$(arg topicname_pose)"/>
        <param name="dataset_name" type="string" value="$(arg dataset_name)"/>
        <param name="rosbag_play" type="bool" value="$(arg rosbag_play)"/>
        <param name="T01_slam" type="bool" value="$(arg T01_slam)"/>
        <param name="data_number" type="string" value="$(arg data_number)"/>
        <param name="dataset_kitti_dir" type="string" value="$(arg dataset_kitti_dir)"/>
        <param name="dataset_carla_dir" type="string" value="$(arg dataset_carla_dir)"/>
        <!-- <param name="/use_sim_time" value="true" /> -->
    </node>

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find mapless_moving)/rviz_cfg/mapless_moving_base.rviz" />
    </group>
    
</launch>